src/libertem/utils/devices.py

```diff
@@ -39,20 +40,30 @@ def detect():
     cores = psutil.cpu_count(logical=False)
     if cores is None:
         cores = 2
-
+    cuda_info = {}
     try:
         cudas = [device.id for device in numba.cuda.gpus]
+        for cuda in cudas:
+            cuda_info[cuda] = {}
+            cuda_ctx = numba.cuda.current_context(cuda)
+            try:
+                cuda_info[cuda]['mem_info'] = cuda_ctx.get_memory_info()
```

Comment:
IMHO we should unpack this into our own data structure, so the interface isn't defined as "whatever numba does currently". We should own the types used in our interfaces. I wouldn't object to copying the current `numba` behavior like this:

```python
class MemoryInfo(typing.NamedTuple):
    free: int
    total: int
```

Reply:
Should be addressed in #1353 , resolving here